### PR TITLE
kubeadm: Restructure internal config usage and fix bugs

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:
 - Moves the generic LoadYAML function from the versioned, external API package to a helper library so it can be consumed more easily
 - Makes the upgrading code use the internal version of the API (which always should be used anyway)
 - Moves all config-loading code to `configutil`, together with the migration code needed. This way we have everything in one centralized place, instead of duplicating that logic N times.
 - Makes `kubeadm init` use `configutil` for the reasons mentioned above.

This PR is needed in order to support multiple external API groups (like v1alpha2)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of https://github.com/kubernetes/community/pull/2131

**Special notes for your reviewer**:
This PR depends on:
 - https://github.com/kubernetes/kubernetes/pull/63782
 - https://github.com/kubernetes/kubernetes/pull/63783

**Please review only the last (third) commit**

**Release note**:

```release-note
NONE
```
@kubernetes/sig-cluster-lifecycle-pr-reviews @liztio

### CODE REVIEW

**Comment by liztio** [comment]
Needs a `./hack/update-bazel.sh` looks like

File: cmd/kubeadm/app/apis/kubeadm/fuzzer/fuzzer_test.go [hunk 1]
```diff
@@ -14,23 +14,26 @@ See the License for the specific language governing permissions and
...
+package fuzzer
 
 import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/api/testing/roundtrip"
 	"k8s.io/apimachinery/pkg/runtime"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	kubeadmfuzzer "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/fuzzer"
```
Comment (timothysc) [inline cmd/kubeadm/app/apis/kubeadm/fuzzer/fuzzer_test.go]: alias seems unnecessary.

File: cmd/kubeadm/app/cmd/config.go [hunk 2]
```diff
@@ -29,7 +29,8 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clientset "k8s.io/client-go/kubernetes"
 	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
-	kubeadmapiext "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha1"
+	kubeadmscheme "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/scheme"
+	kubeadmapiv1alpha1 "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha1"
```
Comment (timothysc) [inline cmd/kubeadm/app/cmd/config.go]: I'd just call it kubeadmapi for future proofing the code in some spots.

Comment (timothysc) [reply cmd/kubeadm/app/cmd/config.go]: Yeah this is a lesson from the client wrapping, avoid version naming, b/c every update is useless churn.

**Review by timothysc — CHANGES_REQUESTED** [review]
Started review but stopped b/c of alias naming. Avoid alias version naming b/c, that will just be extra work in the future with 0 gain.

**Comment by neolit123** [comment]
`upgrade_test.go` needs `LoadYAML()` from the util package now too.

**Comment by liztio** [comment]
@luxas it looks like there's some overlap between this and #63788?

File: cmd/kubeadm/app/apis/kubeadm/v1alpha1/upgrade_test.go [hunk 3]
```diff
@@ -17,37 +17,11 @@ limitations under the License.
...
 import (
-	"io/ioutil"
 	"testing"
 
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/serializer"
 )
 
-const test196 = "testdata/kubeadm196.yaml"
```
Comment (liztio) [inline cmd/kubeadm/app/apis/kubeadm/v1alpha1/upgrade_test.go]: Where'd this test end up?

Comment (luxas) [reply cmd/kubeadm/app/apis/kubeadm/v1alpha1/upgrade_test.go]: Ugh. I moved it to `testdata` but somehow I didn't commit it. I'll send a non-critical PR with that test later

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve

**Pull request merged**